### PR TITLE
refactoring submit execution flow to include submit() function

### PR DESCRIPTION
**Pull request by atravitz**
partially addresses #238 

## Description

Refactoring the `submit` execution to be:

FlowProject.main() -> FlowProject._main_submit() -> FlowProject.submit() -> submit_operations()

## Motivation and Context
See #238. Currently the `submit()` function was not being called by `_main_submit()`, and `_main_submit()` was replicating the logic.

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [ ] New feature
- [x] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

File: flow/project.py [hunk 1]
```diff
@@ -2608,6 +2608,7 @@ def submit(self, bundle_size=1, jobs=None, names=None, num=None, parallel=False,
 
         # Gather all pending operations.
         default_directives = self._get_default_directives()
+        names = args.operation_name if args.operation_name else None
```
Comment (vyasr) [inline flow/project.py:2611]: Why are you using `args` here? The names are already unpacked in the function parameter list.

File: flow/project.py [hunk 2]
```diff
@@ -2622,8 +2623,8 @@ def submit(self, bundle_size=1, jobs=None, names=None, num=None, parallel=False,
                 force=force, walltime=walltime, **kwargs)
 
             if status is not None:  # operations were submitted, store status
-                for op in bundle:
-                    op.set_status(status)
+                for operations in bundle:
+                    operations.set_status(status)
```
Comment (vyasr) [inline flow/project.py:2628]:
```suggestion
                for operation in bundle:
                    operation.set_status(status)

```

File: flow/project.py (hunk 1 above)
Comment (vyasr) [reply flow/project.py:2611]: I see where this came from in `_main_submit`. You should just be able to drop this entirely, it's getting unpacked by passing the args through to `submit`.

File: flow/project.py [hunk 3]
```diff
@@ -3377,20 +3379,7 @@ def _main_submit(self, args):
...
-                                                         args.parallel, args.ignore_conditions,
-                                                         args.ignore_conditions_on_execution)
-        operations = list(islice(operations, args.num))
-        # Bundle operations up, generate the script, and submit to scheduler.
-        for bundle in make_bundles(operations, args.bundle_size):
-            status = self.submit_operations(operations=bundle, **kwargs)
-            if status is not None:
-                for operations in bundle:
-                    operations.set_status(status)
+        submit(jobs=jobs, **kwargs)
```
Comment (vyasr) [inline flow/project.py:3382]:
```suggestion
        self.submit(jobs=jobs, **kwargs)
```

File: flow/project.py (hunk 1 above)
Comment (atravitz) [reply flow/project.py:2611]: oh okay, this is what I was asking about the other day. thanks for clarifying!

**Comment by vyasr** [comment]
@atravitz it looks like you didn't actually apply all my changes, in particular you didn't apply the change to call `self.submit` so it's still failing because there is no known `submit` function.

File: flow/project.py (hunk 1 above)
Comment (atravitz) [reply flow/project.py]: i moved this back to _main_submit. names has to be explicitly passed in because it's named `operation_name` in args

**Comment by mikemhenry** [comment]
@vyasr I think this is ready for another look

**Pull request merged**